sdks/python/apache_beam/utils/annotations_test.py

```diff
@@ -0,0 +1,113 @@
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import unittest
+import warnings
+from annotations import deprecated
+from annotations import experimental
+
+
+class AnnotationTests(unittest.TestCase):
+  # Note: use different names for each of the the functions decorated
+  # so that a warning is produced for each of them.
+  def test_deprecated_with_since_current(self):
+    with warnings.catch_warnings(record=True) as w:
+      @deprecated(since='v.1', current='multiply')
+      def fnc_test_deprecated_with_since_current():
+        return 'lol'
+      fnc_test_deprecated_with_since_current()
+      self.check_annotation(w, 1, DeprecationWarning,
+                            'fnc_test_deprecated_with_since_current',
+                            'deprecated',
+                            [('since', True), ('instead', True)])
+
+  def test_deprecated_without_current(self):
+    with warnings.catch_warnings(record=True) as w:
+      @deprecated(since='v.1')
+      def fnc_test_deprecated_without_current():
+        return 'lol'
+      fnc_test_deprecated_without_current()
+      self.check_annotation(w, 1, DeprecationWarning,
+                            'fnc_test_deprecated_without_current', 'deprecated',
+                            [('since', True), ('instead', False)])
+
+  def test_deprecated_without_since_should_fail(self):
+    with warnings.catch_warnings(record=True) as w:
+      with self.assertRaises(TypeError):
+
+        @deprecated()
+        def fnc_test_deprecated_without_since_should_fail():
+          return 'lol'
+        fnc_test_deprecated_without_since_should_fail()
+      assert len(w) == 0
+
+  def test_experimental_with_current(self):
+    with warnings.catch_warnings(record=True) as w:
+      @experimental(current='multiply')
+      def fnc_test_experimental_with_current():
+        return 'lol'
+      fnc_test_experimental_with_current()
+      self.check_annotation(w, 1, FutureWarning,
+                            'fnc_test_experimental_with_current',
+                            'experimental', [('instead', True)])
+
+  def test_experimental_without_current(self):
+    with warnings.catch_warnings(record=True) as w:
+      @experimental()
+      def fnc_test_experimental_without_current():
+        return 'lol'
+      fnc_test_experimental_without_current()
+      self.check_annotation(w, 1, FutureWarning,
```

Comment:
Since the number of parameters to function check_annotations() is long it might be better to invoke the parameters in the form of keyword arguments here and below. e.g.:   

self.check_annotation(
    warning=w, waring_size=1, warning_type=FutureWarning, ...)